packages/ai/src/ai/web/scheduler/models.py

```
@@ -0,0 +1,74 @@
+    cron_expression: str = Field(..., min_length=9, max_length=256, description='Cron expression (5-field format: minute hour day month weekday).')
+    input_data: Optional[Dict[str, Any]] = Field(default=None, description='Optional input data passed to the pipeline on each run.')
+    name: str = Field(..., min_length=1, max_length=256, description='Human-readable name for this schedule.')
+    enabled: bool = Field(default=True, description='Whether the schedule should be active immediately.')
+
+    @field_validator('cron_expression')
+    @classmethod
+    def validate_cron_expression(cls, v: str) -> str:
+        """Validate that the cron expression has exactly 5 fields."""
+        parts = v.strip().split()
+        if len(parts) != 5:
+            raise ValueError('Cron expression must have exactly 5 fields (minute hour day month weekday)')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)